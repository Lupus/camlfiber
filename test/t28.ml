let main () =
  let open Unix in
  let a, b = socketpair PF_UNIX SOCK_SEQPACKET 0 in
  set_nonblock a;
  set_nonblock b;
  let n = 64 in
  let reader = Fiber.create (fun () ->
      let buf = Bytes.create 256 in
      for i = 1 to n do
	Fiber.wait_io_ready a Fiber.EV_READ;
	let n = read a buf 0 256 in
	print_string "read:  ";
	print_string (Bytes.sub_string buf 0 n);
	print_newline ()
      done) ()
  in
  let writer = Fiber.create (fun () ->
      let buf = Bytes.make 256 'a' in
      for i = 1 to n do
	Fiber.wait_io_ready b Fiber.EV_WRITE;
	let n' = single_write b buf 0 i in
	print_string "write: ";
	print_string (Bytes.sub_string buf 0 n');
	print_newline ()
      done) () in
  Fiber.resume reader;
  Fiber.resume writer;
  Fiber.join reader

let _ = Fiber.run main ()
