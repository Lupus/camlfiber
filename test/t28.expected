write: a
read:  a
write: aa
read:  aa
write: aaa
read:  aaa
write: aaaa
read:  aaaa
write: aaaaa
read:  aaaaa
write: aaaaaa
read:  aaaaaa
write: aaaaaaa
read:  aaaaaaa
write: aaaaaaaa
read:  aaaaaaaa
write: aaaaaaaaa
read:  aaaaaaaaa
write: aaaaaaaaaa
read:  aaaaaaaaaa
write: aaaaaaaaaaa
read:  aaaaaaaaaaa
write: aaaaaaaaaaaa
read:  aaaaaaaaaaaa
write: aaaaaaaaaaaaa
read:  aaaaaaaaaaaaa
write: aaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
write: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
read:  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
